nodejs/eks/examples/migrate-nodegroups/steps/step3/index.ts

```diff
@@ -0,0 +1,89 @@
+import * as awsx from "@pulumi/awsx";
```

Comment:
It's hard to tell what is changing between these steps.  That's related to the overall complexity of the test case that seems unrelated to what it is trying to test.  Could we either simplify this down to its essence, or at least factor out some of the common code here that doesn't change into a separate so the changes are clearer?

Reply:
In hopes to call out the over-arching steps:

- step 0:
  - stand up cluster, standard ng, 2xlarge node group, and workload (echoserver + nginx)
- step 1:
  - stand up new 4xlarge ng
- step 2:
  - retarget nginx's nodeSelector from 2xlarge -> 4xlarge
- step 3:
  - decommission 2xlarge ng (drain & delete)
  - delete 2xlarge ng

I thought each step in its own testing step overwrite would make the most sense, and the deltas are also < 10 lines each this way. Any alternatives stand out to you?

Reply:
There’s just so much other code in this file that is getting duplicated. Just feels like a lot of incidental complexity unrelated to the task at hand.

Not a blocker if you think this is right - but this feels likely to be a difficult to maintain test.

Reply:
To further help, below is a visual representation of the test steps from the [tutorial PR](https://github.com/pulumi/docs/pull/1327) - they were split into the current form in the tests, as this is the flow we built and use in the migration tutorial and blog post.

I'm going to leave the existing example & steps as is in order to support the working tutorial and blog post, but I'll see what I can come up with for a test that is less complex.

Are your complexity concerns primarily for the test setup & dupe code, or for the whole topic of the example too?

### **Step 1:**

![image](https://user-images.githubusercontent.com/1112768/61259306-78e33080-a72e-11e9-95fe-3ead858d55f5.png)

### **Step 2:**
![image](https://user-images.githubusercontent.com/1112768/61259312-83052f00-a72e-11e9-82d0-145cdae1dfde.png)

### **Step 3:**
![image](https://user-images.githubusercontent.com/1112768/61259316-8698b600-a72e-11e9-8245-3714061185b5.png)

Reply:
Honestly - if it’s easiest to just keep this test as is - I’m fine with that. We can pare back inessential complexity here if and only if we see brittleness around this test in the future instead of creating new work now. 

I *love* this test case for highlighting the full end-to/end user experience!

Reply:
Tests have been adapted. See [comments](https://github.com/pulumi/pulumi-eks/pull/195#issuecomment-511995592).